lib/optimize/SplitChunksPlugin.js

```diff
@@ -305,15 +305,17 @@ module.exports = class SplitChunksPlugin {
 								if (chunkIndices.length < cacheGroup.minChunks) continue;
 								// Select chunks by configuration
 								const selectedChunks =
-									cacheGroup.chunks === "initial"
-										? Array.from(chunkCombination).filter(chunk =>
-												chunk.canBeInitial()
-											)
-										: cacheGroup.chunks === "async"
-											? Array.from(chunkCombination).filter(
-													chunk => !chunk.canBeInitial()
+									typeof cacheGroup.chunks === "function"
```

Comment:
This is getting too big. Move the "constant string to filter" logic into a separate function and use ifs instead of ?:

Reply:
By the way I'm not sure there's a sense in supporting this option in selector (`Array<Chunk> => Array<Chunk>`) form. Maybe it's enough to support predicate (`Chunk => boolean`)?

Reply:
> "constant string to filter" logic

Should it be a part of `normalizeOptions` maybe?

Reply:
Yes that makes sense.

The normalization could normalize the cache group options to a `chunkSelector` filter function. `"initial"`, `"all"` and `"async"` can be normalized to functions too.

Here we only have to use `Array.from(chunkCombination).filter(cacheGroup.chunkSelector)`.

Reply:
Maybe put the logic into a `normalizeChunks` helper, because it's used twice in `normalizeCacheGroups`, similar to `normalizeName`